helm/cluster-vsphere/templates/vspheremachinetemplate.yaml

```diff
@@ -1,4 +1,17 @@
-{{- range $nodePoolName, $value := include "createMapOfClusterNodeSpecs" . | fromYaml }}
+---
+apiVersion: {{ include "infrastructureApiVersion" . }}
+kind: VSphereMachineTemplate
+metadata:
+  name: {{ include "resource.default.name" $ }}-control-plane-{{ include "machineTemplateSpec.hash" (dict "data" (include "controlplane-vspheremachinetemplate-spec" $) "salt" $.Values.cluster.providerIntegration.hashSalt) }}
+  namespace: {{ $.Release.Namespace }}
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+spec:
+  template:
+    spec:
+      {{- include "controlplane-vspheremachinetemplate-spec" $ | nindent 6 -}}
+
```

Comment:
As the `cluster` chart uses different methods to create the name of the machine resources, we need to align methods. Previously, this template used a function which could range over both CP and nodepool blocks, however whilst the `cluster` chart is only partly integrated the two naming schemes are incompatible. So we now have an individual `vspheremachinetemplate` for each set of nodes.